extensions/authentication-jwt/src/role-bases.authorizor.ts

```diff
@@ -0,0 +1,75 @@
+// Copyright IBM Corp. 2020. All Rights Reserved.
+// Node module: @loopback/authentication-jwt
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  AuthorizationContext,
+  AuthorizationDecision,
+  AuthorizationMetadata,
+} from '@loopback/authorization';
+import {securityId, UserProfile} from '@loopback/security';
+import _ from 'lodash';
+
+// Instance level authorizer
+// Can be also registered as an authorizer, depends on users' need.
+export async function roleBasedAuthorization(
+  authorizationCtx: AuthorizationContext,
+  decoratorMetadata: AuthorizationMetadata,
+): Promise<AuthorizationDecision> {
+  // No access if authorization details are missing
+  let currentUser: UserProfile;
+  if (authorizationCtx.principals.length > 0) {
+    const user = _.pick(authorizationCtx.principals[0], [
+      'id',
+      'name',
+      'email',
+      'roles',
+    ]);
+    currentUser = {
+      [securityId]: user.id,
+      name: user.name,
+      email: user.email,
+      roles: user.roles,
+    } as UserProfile;
+  } else {
+    return AuthorizationDecision.DENY;
+  }
```

Comment:
Let's reduce the nesting by inverting the condition:

```suggestion
  if (authorizationCtx.principals.length <= 0) {
    return AuthorizationDecision.DENY;
  }
  const user = _.pick(authorizationCtx.principals[0], [
    'id',
    'name',
    'email',
    'roles',
  ]);
  currentUser = {
    [securityId]: user.id,
    name: user.name,
    email: user.email,
    roles: user.roles,
  } as UserProfile;
```